### DISCUSSION TITLE
DC parameter sweep in ngspice

### DISCUSSION THREAD

**Discussion by iwbnwif** [ra3xdh/qucs_s#577] [Q&A] [answered]
I might be wrong, but I am sure it used to be possible to make a DC parameter sweep using a setup like the following:
![image](https://github.com/ra3xdh/qucs_s/assets/11036083/95c5c663-6071-438d-a975-8cd6e0d99cd8)
But this now fails with:
 ```
Fatal error: DC Transfer Function: Voltage source, current source, or resistor named "@d1[vj]" is not in the circuit
doAnalyses: no such device
```

This is caused by Qucs-S creating a netlist with the following instead of a loop:

`dc @d1[vj] 0.5 0.9 0.08`

Is this intentional, and if so, how should I run a DC parameter sweep?

**Comment by ra3xdh** [comment]
> Is this intentional, and if so, how should I run a DC parameter sweep?

The loop is not required for DC sweep. Furthermore loop will produce a set of 1x1 vectory and will not work here. It works differently. See the section 15.3.2 of the Ngspice manual for allowed syntax. DC sweep doesn't allow variables and device parameters.  Please report to the Ngspice forum to request feature extension.

**Comment by iwbnwif** [comment, marked as answer]
Thank you for you answer.

>  Please report to the Ngspice forum to request feature extension.

Yes, that would be a nice feature. If I understand the documentation correctly, Xyce already has this feature.

In the meantime, I think I can sweep the operating point instead using a Nutmeg script:

![image](https://github.com/ra3xdh/qucs_s/assets/11036083/b58a0ee0-cd50-45d0-942d-30f702af911b)

I would be happy to receive any comments on this script and/or approach.
